src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -142,6 +149,14 @@ else if (currentState != SyncthingService.State.ACTIVE) {
             }
             mDeviceId.setOnPreferenceClickListener(this);
         }
+
+        if (mDevice == null) {
+            // The device was removed from Syncthing and no reference was found
+            Log.d(TAG, "Empty mDevice");
+            getActivity().finish();
+            return;
+        }
```

Comment:
This got me thinking, mDevice is updated by a postExecuted task, so I guess this can happen at any time, correct?
We should use a local reference (i.e. device = mDevice) and then check if it is null and use the device reference in the remaining code.